lib/libvarnish/vre.c

```diff
@@ -200,9 +203,24 @@ vre_match(const vre_t *code, const char *subject, size_t length, size_t offset,
 		AN(data);
 	}
 
-	matches =  pcre2_match(re, (PCRE2_SPTR)subject, length, offset,
+	matches = pcre2_match(re, (PCRE2_SPTR)subject, length, offset,
 	    options, data, code->re_ctx);
 
+	if (groups != NULL) {
+		AN(count);
+		AN(*count);
+		ovector = pcre2_get_ovector_pointer(data);
+		nov = pcre2_get_ovector_count(data);
+		if (nov > *count)
+			nov = *count;
+		for (g = 0; g < nov; g++) {
+			groups->b = subject + ovector[2 * g];
+			groups->e = subject + ovector[2 * g + 1];
+			groups++;
+		}
```

Comment:
Should we `NULL` remaining groups (for `nov < *count`) or leave that to the caller?

Reply:
I wouldn't bother, but that may be my laziness speaking on my behalf.